tests/check_mlflow_lazily_imports_ml_packages.py

```diff
@@ -0,0 +1,50 @@
+"""
+Tests that mlflow doesn't import ml packages when it's imported
+"""
+
+import sys
+import importlib
+import mlflow  # pylint: disable=unused-import
+
+
+def main():
+    ml_packages = {
+        "catboost",
+        "fastai",
+        "mxnet",
+        "h2o",
+        "keras",
+        "lightgbm",
+        "mleap",
+        "onnx",
+        "pytorch_lightning",
+        "pyspark",
+        "shap",
+        "sklearn",
+        "spacy",
+        "statsmodels",
+        "tensorflow",
+        "torch",
+        "xgboost",
+        "pmdarima",
+        "diviner",
+    }
+    imported = ml_packages.intersection(set(sys.modules))
+    assert imported == set(), f"mlflow imports {imported} when it's imported but it should not"
+
+    # Ensure that the ML packages are importable
+    failed_to_import = []
+    for package in ml_packages:
+        try:
+            importlib.import_module(package)
+        except ImportError:
+            failed_to_import.append(package)
+
+    message = (
+        f"Failed to import {failed_to_import}. Please install packages that provide these modules."
+    )
+    assert failed_to_import == [], message
```

Comment:
We can add a similar test, to check after running `import mlflow; mlflow.autolog()`, all these packages are not imported except pyspark.

Reply:
Sounds good. I'll add it.

Reply:
Done